### PR TITLE
Processed Products Sales Computing method

### PR DESCRIPTION
**Pull request by dm-ps**
refs: https://github.com/metasfresh/metasfresh/issues/17983

### CODE REVIEW

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/modular/interceptor/PP_Order.java [hunk 1]
```diff
@@ -97,4 +103,16 @@ public void onBeforeComplete(@NonNull final I_PP_Order order)
 			modularPPOrderService.validateModularOrder(ppOrderId);
 		}
 	}
+
+	@DocValidate(timings = ModelValidator.TIMING_AFTER_COMPLETE)
+	public void afterComplete(@NonNull final I_PP_Order order)
```
Comment (teosarca) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/modular/interceptor/PP_Order.java:108]: as discussed, consider 
* either triggering this event of after close
* either triggering this event on PP_Cost_Collector of type receipt

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/modular/interceptor/PP_Order.java (hunk 1 above)
Comment (dm-ps) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/modular/interceptor/PP_Order.java]: went with "either triggering this event on PP_Cost_Collector of type receipt"

File: backend/de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5723210_sys_gh17983_ProcessedSalesComputingMethod.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
+-- Run mode: SWING_CLIENT
+
+-- Name: ModCntr_Type
+-- 2024-05-08T08:51:14.402Z
+INSERT INTO AD_Val_Rule (AD_Client_ID,AD_Org_ID,AD_Val_Rule_ID,Code,Created,CreatedBy,EntityType,IsActive,Name,Type,Updated,UpdatedBy) VALUES (0,0,540674,'(CASE WHEN @M_Product_ID/-1@ != @M_Processed_Product_ID/-1@ AND modcntr_type.modularcontracthandlertype IN (''SalesOnProcessedProduct'', ''AddValueOnProcessedProduct'') THEN false
```
Comment (teosarca) [inline backend/de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5723210_sys_gh17983_ProcessedSalesComputingMethod.sql:5]: pls give a better name to this validation rule.

Comment (teosarca) [reply backend/de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5723210_sys_gh17983_ProcessedSalesComputingMethod.sql:5]: about the SQL logic, isn't it
```sql
@M_Product_ID/-1@ = @M_Processed_Product_ID/-1@ OR modcntr_type.modularcontracthandlertype NOT IN (''SalesOnProcessedProduct'',''AddValueOnProcessedProduct'') 
```

**Pull request merged**